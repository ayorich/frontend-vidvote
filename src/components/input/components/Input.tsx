import React, { ReactElement, FC } from 'react';
import { Input as AntInput } from 'antd';
import { inputProps } from '../types';

const getComponenByType = (componenctType: string | undefined) => {
    switch (componenctType) {
        case 'Search':
            return AntInput['Search'];
        case 'Password':
            return AntInput['Password'];
        case 'TextArea':
            return AntInput['TextArea'];
        default:
            return AntInput;
    }
};

const CustomInput: FC<inputProps> = ({
    label,
    placeholder,
    hasLabel,
    onChange,
    componentType,
    inputType,
    name,
    value,
    disabled,
}): ReactElement => {
    const Input = getComponenByType(componentType);
    return (
        <div className="custom-input">
            {hasLabel && <label htmlFor={name}>{label}</label>}
            <Input
                disabled={disabled}
                value={value}
                id={name}
                name={name}
                type={inputType}
                placeholder={placeholder}
                onChange={onChange}
            />
        </div>
    );
};

export default CustomInput;
