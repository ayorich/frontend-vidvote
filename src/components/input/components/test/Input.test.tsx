import React from 'react';
import Input from '../Input';
import { render } from '@testing-library/react';

test('render Input component', () => {
    const { asFragment } = render(<Input />);
    expect(asFragment()).toMatchSnapshot();
});
