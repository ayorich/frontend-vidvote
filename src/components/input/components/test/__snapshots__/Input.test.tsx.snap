// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Input component 1`] = `
<DocumentFragment>
  <div
    class="custom-input"
  >
    <input
      class="ant-input"
      type="text"
      value=""
    />
  </div>
</DocumentFragment>
`;
